Implicit fallthrough warning on GCC 7.x

I have git master up-to-date as at 2017-02-14T08:13+00:00, with Fedora Rawhide fully up to date, and attempt to build Pony results in:
```
|> make
Makefile:224: WARNING: Unsupported LLVM version: 3.9.0
Makefile:225: Please use LLVM 3.7.1, 3.8.1, or 3.9.1
actor.c
messageq.c
fun.c
src/libponyrt/ds/fun.c: In function ‘siphash24’:
src/libponyrt/ds/fun.c:46:15: error: this statement may fall through [-Werror=implicit-fallthrough=]
     case 7: b |= ((uint64_t)in[6]) << 48;
             ~~^~~~~~~~~~~~~~~~~~~~~~~~~~
src/libponyrt/ds/fun.c:47:5: note: here
     case 6: b |= ((uint64_t)in[5]) << 40;
     ^~~~
src/libponyrt/ds/fun.c:47:15: error: this statement may fall through [-Werror=implicit-fallthrough=]
     case 6: b |= ((uint64_t)in[5]) << 40;
             ~~^~~~~~~~~~~~~~~~~~~~~~~~~~
src/libponyrt/ds/fun.c:48:5: note: here
     case 5: b |= ((uint64_t)in[4]) << 32;
     ^~~~
src/libponyrt/ds/fun.c:48:15: error: this statement may fall through [-Werror=implicit-fallthrough=]
     case 5: b |= ((uint64_t)in[4]) << 32;
             ~~^~~~~~~~~~~~~~~~~~~~~~~~~~
src/libponyrt/ds/fun.c:49:5: note: here
     case 4: b |= ((uint64_t)in[3]) << 24;
     ^~~~
src/libponyrt/ds/fun.c:49:15: error: this statement may fall through [-Werror=implicit-fallthrough=]
     case 4: b |= ((uint64_t)in[3]) << 24;
             ~~^~~~~~~~~~~~~~~~~~~~~~~~~~
src/libponyrt/ds/fun.c:50:5: note: here
     case 3: b |= ((uint64_t)in[2]) << 16;
     ^~~~
src/libponyrt/ds/fun.c:50:15: error: this statement may fall through [-Werror=implicit-fallthrough=]
     case 3: b |= ((uint64_t)in[2]) << 16;
             ~~^~~~~~~~~~~~~~~~~~~~~~~~~~
src/libponyrt/ds/fun.c:51:5: note: here
     case 2: b |= ((uint64_t)in[1]) << 8;
     ^~~~
src/libponyrt/ds/fun.c:51:15: error: this statement may fall through [-Werror=implicit-fallthrough=]
     case 2: b |= ((uint64_t)in[1]) << 8;
             ~~^~~~~~~~~~~~~~~~~~~~~~~~~
src/libponyrt/ds/fun.c:52:5: note: here
     case 1: b |= ((uint64_t)in[0]);
     ^~~~
cc1: all warnings being treated as errors
```

--- a/src/libponyrt/ds/fun.c
+++ b/src/libponyrt/ds/fun.c
@@ -43,12 +43,12 @@ static uint64_t siphash24(const unsigned char* key, const char* in, size_t len)
 
   switch(len & 7)
   {
-    case 7: b |= ((uint64_t)in[6]) << 48;
-    case 6: b |= ((uint64_t)in[5]) << 40;
-    case 5: b |= ((uint64_t)in[4]) << 32;
-    case 4: b |= ((uint64_t)in[3]) << 24;
-    case 3: b |= ((uint64_t)in[2]) << 16;
-    case 2: b |= ((uint64_t)in[1]) << 8;
+    case 7: b |= ((uint64_t)in[6]) << 48; // -fallthrough
+    case 6: b |= ((uint64_t)in[5]) << 40; // -fallthrough
+    case 5: b |= ((uint64_t)in[4]) << 32; // -fallthrough
+    case 4: b |= ((uint64_t)in[3]) << 24; // -fallthrough
+    case 3: b |= ((uint64_t)in[2]) << 16; // -fallthrough
+    case 2: b |= ((uint64_t)in[1]) << 8;  // -fallthrough
     case 1: b |= ((uint64_t)in[0]);
   }
 

